packages/web/src/components/runs-table.tsx

```diff
@@ -48,12 +66,29 @@ interface RunsTableProps {
  * which fetches all data upfront and paginates client-side.
  */
 export function RunsTable({ config, onRunClick }: RunsTableProps) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const status = searchParams.get('status') as
+    | WorkflowRunStatus
+    | 'all'
+    | undefined;
+  const workflowNameFilter = searchParams.get('workflow') as string | 'all';
   const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('desc');
+
+  // TODO: This is a workaround. We should be getting a list of valid workflow names
+  // from the manifest, which we need to put on the World interface.
+  const [seenWorkflowNames, setSeenWorkflowNames] = useState<Set<string>>(
+    new Set()
+  );
   const [lastRefreshTime, setLastRefreshTime] = useState<Date | null>(
     () => new Date()
   );
   const env = useMemo(() => worldConfigToEnvMap(config), [config]);
 
+  const statusFilterRequiresWorkflowNameFilter =
+    config.backend?.includes('vercel');
```

Comment:
oof. If this is a must, let's please add a TODO comment atleast so we're not adding world specifics to agnostic packages

this should be standardized in the world spec